apps/designer/app/designer/features/style-panel/shared/property-name.tsx

```diff
@@ -1,39 +1,98 @@
+import { forwardRef, PointerEventHandler } from "react";
 
 export const PropertyName = ({ property, label, css }: PropertyProps) => {
   const isCurrentBreakpoint = useIsFromCurrentBreakpoint(property);
 
   return (
-    <Flex
-      css={{
-        ...css,
-        alignItems: "center",
-      }}
-    >
-      <Label
+    <Tooltip content={label} delayDuration={600} disableHoverableContent={true}>
+      <Flex
         css={{
-          gridColumn: "1",
-          fontWeight: "inherit",
-          ...(isCurrentBreakpoint
-            ? {
-                color: propertyNameColorForSelectedBreakpoint,
-                backgroundColor: "$colors$blue4",
-                padding: "calc($radii$1 / 2) $radii$1",
-                borderRadius: "$radii$1",
-              }
-            : {
-                color: "$hiContrast",
-              }),
+          ...css,
+          alignItems: "center",
         }}
-        variant="contrast"
-        size="1"
-        htmlFor={property}
       >
-        {label}
-      </Label>
-    </Flex>
+        <Label
+          css={{
+            gridColumn: "1",
+            fontWeight: "inherit",
+            ...(isCurrentBreakpoint
+              ? {
+                  color: propertyNameColorForSelectedBreakpoint,
+                  backgroundColor: "$colors$blue4",
+                  padding: "calc($radii$1 / 2) $radii$1",
+                  borderRadius: "$radii$1",
+                }
+              : {
+                  color: "$hiContrast",
+                }),
+          }}
+          variant="contrast"
+          size="1"
+          htmlFor={property}
+        >
+          {label}
+        </Label>
+      </Flex>
+    </Tooltip>
   );
 };
+PropertyName.displayName = "PropertyName";
+
+export const PropertyIcon = forwardRef<
+  HTMLButtonElement,
+  PropertyProps & { [key: `on${string}`]: PointerEventHandler }
+>(({ property, label, ...props }, forwardRef) => {
+  const isCurrentBreakpoint = useIsFromCurrentBreakpoint(property);
+  const IconType = iconConfigs[property]?.normal;
+  if (!IconType) return null;
+
+  return (
+    <Tooltip content={label} delayDuration={600} disableHoverableContent={true}>
+      <IconButton
+        {...props}
+        ref={forwardRef}
+        variant="ghost"
+        size="1"
+        css={{
+          position: "absolute",
+          left: "1px",
+          top: "1px",
+          width: "$5",
+          height: "calc(100% - 2px)",
+          borderRadius: "$1",
+          border: "2px solid $colors$loContrast",
+          "&:hover,&:active,&:focus": {
+            bc: "$colors$slate4",
+            outline: "none",
+          },
+          ...(isCurrentBreakpoint && {
+            bc: "$colors$blue4",
+            color: "$colors$blue11",
+            outline: "none",
+            "&:hover,&:active,&:focus": {
+              bc: "$colors$blue4",
+              color: "$colors$blue11",
+            },
+          }),
+          [`& ~ ${TextField}`]: {
```

Comment:
this shouldn't be necessary, it basically introduces knowledge of a TextField next to a property name component, breaks encapsulation again

Reply:
Why is this needed?

Reply:
I think @okonet just created a PR for a text field that allows rendering anything on the left or the right side of the input, so that TextField is in charge of it's layout and we don't have to position it from the outside.